impeller/entity/entity_pass.h

```diff
@@ -52,6 +52,13 @@ class EntityPass {
 
   void IterateAllEntities(const std::function<bool(Entity&)>& iterator);
 
+  /// @brief Iterates all entities on this pass, returning if there is a
+  /// subpass.
+  bool IterateAllFlatEntities(const std::function<bool(Entity&)>& iterator);
```

Comment:
Small nit, but maybe rename to "IterateUntilSubpass"?

Reply:
Done